modules/readpeakBidAdapter.js

```diff
@@ -0,0 +1,237 @@
+import {logError, getTopWindowLocation} from 'src/utils';
+import { registerBidder } from 'src/adapters/bidderFactory';
+
+export const ENDPOINT = '//app.readpeak.com/header/prebid';
+
+const NATIVE_DEFAULTS = {
+  TITLE_LEN: 70,
+  DESCR_LEN: 120,
+  SPONSORED_BY_LEN: 50,
+  IMG_MIN: 150,
+  ICON_MIN: 50,
+  CTA_LEN: 50,
+};
+
+const BIDDER_CODE = 'readpeak'
+
+export const spec = {
+
+  code: BIDDER_CODE,
+
+  supportedMediaTypes: ['native'],
+
+  isBidRequestValid: bid => (
+    !!(bid && bid.params && bid.params.publisherId && bid.nativeParams)
+  ),
+
+  buildRequests: bidRequests => {
+    const request = {
+      id: bidRequests[0].bidderRequestId,
+      imp: bidRequests.map(slot => impression(slot)).filter(imp => imp.native != null),
+      site: site(bidRequests),
+      app: app(bidRequests),
+      device: device(),
+      isPrebid: true,
+    }
+
+    return {
+      method: 'POST',
+      url: ENDPOINT,
+      data: JSON.stringify(request),
+    }
+  },
+
+  interpretResponse: (response, request) => (
+    bidResponseAvailable(request, response)
+  ),
+};
+
+function bidResponseAvailable(bidRequest, bidResponse) {
+  const idToImpMap = {};
+  const idToBidMap = {};
+  if (!bidResponse['body']) {
+    return []
+  }
+  bidResponse = bidResponse.body
+  parse(bidRequest.data).imp.forEach(imp => {
+    idToImpMap[imp.id] = imp;
+  });
+  if (bidResponse) {
+    bidResponse.seatbid.forEach(seatBid => seatBid.bid.forEach(bid => {
+      idToBidMap[bid.impid] = bid;
+    }));
+  }
+  const bids = [];
+  Object.keys(idToImpMap).forEach(id => {
+    if (idToBidMap[id]) {
+      const bid = {
+        requestId: id,
+        cpm: idToBidMap[id].price,
+        creativeId: id,
+        adId: id,
```

Comment:
`adId` can be dropped. I forgot this was the legacy bid identifier. It will be found on the `requestId` now